src/gui/socketapi.cpp

```diff
@@ -717,9 +724,18 @@ void SocketApi::command_GET_MENU_ITEMS(const QString &argument, OCC::SocketListe
     FileData fileData = hasSeveralFiles ? FileData{} : FileData::get(argument);
     bool isOnTheServer = fileData.journalRecord().isValid();
     auto flagString = isOnTheServer ? QLatin1String("::") : QLatin1String(":d:");
+    auto capabilities = fileData.folder->accountState()->account()->capabilities();
+
     if (fileData.folder && fileData.folder->accountState()->isConnected()) {
         sendSharingContextMenuOptions(fileData, listener);
         listener->sendMessage(QLatin1String("MENU_ITEM:OPEN_PRIVATE_LINK") + flagString + tr("Open in browser"));
+
+        QMimeDatabase db;
+        QMimeType type = db.mimeTypeForFile(fileData.localPath);
+        if (capabilities.hasRichDocuments() && capabilities.supportedRichDocumentsMimetypes().contains(type.name().toLatin1())){
+            listener->sendMessage(QLatin1String("MENU_ITEM:EDIT") + flagString + tr("Edit via ") + capabilities.richDocumentsProductName());
```

Comment:
Just to clear this up, we only want it to say "Edit", right? No product name, no "… collaboratively". Can you confirm @karlitschek?

cc @camilasan fyi

Reply:
(Also, "Edit" should be the first in the list, before the Sharing entries – because you can also do that when the document is not shared.)

Reply:
@jancborchardt Yes. I would only say 'Edit'. The app that is used depends on the mime type and is not really important for the user.

Reply:
```suggestion
            listener->sendMessage(QLatin1String("MENU_ITEM:EDIT") + flagString + tr("Edit"));
```

Reply:
While implementing this feature with OO, we found a small "problem": you can open also pdf files, which are not editable.
So we thought about "open (remotely)" instead of "edit" as this is more generic.

Reply:
@tobiasKaminsky can’t we just exclude PDF from it instead? Then the entry will simply say "Open in browser" as specified at https://github.com/nextcloud/desktop/pull/1442#issuecomment-546722480

That’s much better than complicating the wording for the regular case.